### PR TITLE
Update the basic section of the role

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
Update the basic section of the role

### Screenshot
Screenshot 2023-10-23 at 14 38 32

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/roles/api/roles.ts [hunk 1]
```diff
@@ -98,6 +99,41 @@ export const getRoleById = (roleId: string): Promise<any> => {
         });
 };
 
+/**
+ * Retrieve Role details for a give role id.
```
Comment (dasuni-30) [inline apps/console/src/features/roles/api/roles.ts:103]:
```suggestion
 * Retrieve Role details for a given role id.
```

**Review by DonOmalVindula — APPROVED** [review]

File: apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com).
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx:2]: Let's update the license header

File: apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    const {
+        isLoading,
+        roleObject,
+        onRoleUpdate,
+        defaultActiveIndex
+    } = props;
+
+    const { t } = useTranslation();
+
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx:70]: Shall we add nullable checks

File: apps/console/src/features/roles/pages/role-edit-updated.tsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+    const {
+        data: roleObject,
+        isLoading: isRoleDetailsRequestLoading,
+        error: roleDetailsRequestError,
+        mutate: mutateRoleObject
+    } = useGetRoleById(roleId);
+
+    /**
+     * The following useEffect is used to handle if any error occurs while fetching the role details.
```
Comment (dasuni-30) [inline apps/console/src/features/roles/pages/role-edit-updated.tsx:74]: Shall we update the comment
```suggestion
     * Handle if any error occurs while fetching the role details.
```

Comment (Achintha444) [reply apps/console/src/features/roles/pages/role-edit-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4290/commits/40c5244008b0c41e56b06852cec9b531cdf7b110

File: apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx (hunk 3 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4290/commits/40c5244008b0c41e56b06852cec9b531cdf7b110

File: apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx (hunk 2 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4290/commits/40c5244008b0c41e56b06852cec9b531cdf7b110

File: apps/console/src/features/roles/api/roles.ts (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/roles/api/roles.ts]: Fixed with https://github.com/wso2/identity-apps/pull/4290/commits/40c5244008b0c41e56b06852cec9b531cdf7b110

**Review by DonOmalVindula — APPROVED** [review]

**Review by dasuni-30 — APPROVED** [review]

**Pull request merged**